benchmarks/mmmu/README.md

```diff
@@ -0,0 +1,27 @@
+# MMMU: A Massive Multi-discipline Multimodal Understanding and Reasoning Benchmark for Expert AGI
+
+[MMMU](https://arxiv.org/abs/2311.16502) is a benchmark for evaluating multi-modal models on college-level tasks across a variety of subjects. 11.5k problems are gathered across 30 subjects and image types. Around 93% of questions in the evaluation dataset are multiple-choice, with the remainder being open-ended. GPT-4o is reported to achieve 63-69% accuracy for MMMU.
+
+## Execution
+Here is an example from the dataset:
+```
+Question: The double-tiered columns allowed for all of the following EXCEPT (<image 1>)
+Option:
+A) barrel-vaulted roofing
+B) decorative rhythm and repetition
+C) a higher roof to make up for the short columns
+D) the entrance of light and air into the hall
+```
+![alt text](image.png)
+
+The model is required to choose the correct answer from the given options and attached image. In this case, the correct answer is A) barrel-vaulted roofing.
+
+## Evaluation
+All evaluation is zero shot. For multiple-choice questions, one question, four options and up to seven images are provided. For open-ended questions, the four options are omitted. [Prompts](https://github.com/MMMU-Benchmark/MMMU/blob/main/eval/configs/llava1.5.yam) follow those used by the original authors. [Micro-averaged accuracy](https://github.com/MMMU-Benchmark/MMMU/blob/main/eval/utils/eval_utils.py#L245) is used in the original paper, while this implementation simply returns Inspect's accuracy metric for each of the multiple-choice and open-ended tasks.
+
+## TODO
+Implement micro-averaged accuracy (calculate average accuracy per subject, multiply by (number of samples / total samples), sum weighted accuracies). Although, for the validation set, all subjects have the same number of samples.
```

Comment:
**OpenAI:** 

- Seems MMMU, and other multi-modal evals haven’t been [added](https://github.com/openai/simple-evals/commit/e29accf208d88fb1cd4d3b401dc14c3d894a246e) to simple-evals, which makes it difficult to judge this.

**Helm:** 

- There are JSONs available for the per-subject [runs](https://crfm.stanford.edu/helm/vhelm/latest/#/runs/mmmu:subject=Art_Theory,question_type=multiple-choice,model=openai_gpt-4o-2024-05-13), manually summing the accuracy values (expected match) per subject, and taking the mean, gives us their reported accuracy (63.1%). This would indicate to me that they don’t use micro-averaged accuracy. 
- To be completely sure, I checked whether there’s any micro-averaging logic for their accuracy metric (exact match). In their JSON, exact match looks to be a `BasicGenerationMetric` instance, which inherits from this [Metric](https://github.com/stanford-crfm/helm/blob/bbdb00e579f7dd475a54bb8e5d8b75f5e63f9aad/src/helm/benchmark/metrics/metric.py#L99) base class. I see that a mean over trials (one trial per subject) is taken, with no obvious weighting for number of questions (known as instances in Helm).
- Note: they’ve also [filtered out open-ended questions](https://github.com/stanford-crfm/helm/blob/bbdb00e579f7dd475a54bb8e5d8b75f5e63f9aad/src/helm/benchmark/presentation/run_entries_vhelm_lite.conf), and different subjects have different numbers of open-ended questions. 

**Open-Compass:** 

- I haven't found any micro-averaging logic in Open-Compass/VLMEvalKit, but I could have missed something. Given they convert open questions to multiple-choice, as opposed to omitting them, it would seem that results should be the same either way (same number of questions per topic). 
- Note: open-ended questions [reformulated](https://github.com/open-compass/VLMEvalKit/blob/dfe0f697aab4a56351bf781b6676624e922095e8/vlmeval/dataset/utils/multiple_choice.py#L61) to multiple-choice. 
- Note: results vary with version and image detail level: https://huggingface.co/spaces/opencompass/open_vlm_leaderboard